pkg/kv/kvserver/replica_rangefeed.go

```diff
@@ -906,6 +961,16 @@ func (r *Replica) handleClosedTimestampUpdateRaftMuLocked(
 				defer func() { <-m.RangeFeedSlowClosedTimestampNudgeSem }()
 				if err := r.ensureClosedTimestampStarted(ctx); err != nil {
 					log.Infof(ctx, `RangeFeed failed to nudge: %s`, err)
+				} else if shouldCancelTooFarBehind {
+					log.Infof(ctx,
+						`RangeFeed is too far behind, cancelling [behind=%v behind_threshold=%v behind_start=%v behind_duration_threshold=%v]`,
+						behind, extremelySlowLagThresh, extremelyFarBehindStartTime, extremelySlowCloseThresh)
+					// TODO(kvoli): We should be using a new retry error reason, which
+					// has similar retry behavior to the replica removed reason, but is
+					// more specific to this case.
+					//
+					// See https://github.com/cockroachdb/cockroach/blob/b4280f7c6613f362791478696adbdf2cc67dd874/pkg/kv/kvclient/kvcoord/dist_sender_rangefeed.go#L555-L555.
+					r.disconnectRangefeedWithReason(kvpb.RangeFeedRetryError_REASON_REPLICA_REMOVED)
```

Comment:
Good catch, we definitely should. I'll add it.

Reply:
Also, note that we won't enter this code path if the range is unavailable (no leaseholder), because the above RPC this is under will return an error and not enter this conditional.